src/main/java/core/basesyntax/data/ReportGenerator.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax.data;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class ReportGenerator {
+    public List<String> generateReport(Map<String, Integer> inventory) {
+        if (validateInputMap(inventory)) {
+            throw new IllegalArgumentException(
+                    "Error while generate report. Map cannot be null or empty!");
+        }
+        List<String> result = new ArrayList<>();
+        result.add("fruit,quantity");
+        for (String key : inventory.keySet()) {
+            result.add(key + "," + inventory.get(key));
+        }
+        inventory.forEach((key, value) -> result.add(key + "," + value));
```

Comment:
The loop on line 15 and the `forEach` method on line 18 both add entries to the result list, resulting in duplicate entries in the report. Consider removing one of these to avoid redundancy.